### ISSUE TITLE
Brackets reset default language to plain/text

### ISSUE THREAD

**Issue by Athorcis** [adobe/brackets#12970]
### Prerequisites

* [ ] Can you reproduce the problem with `Debug -> Reload Without Extensions`?
Only because this bug is linked to highlighting extensions.
* [x] Did you perform a cursory search to see if your bug or enhancement is already reported?
* [x] Did you read the [Troubleshooting guide](https://github.com/adobe/brackets/wiki/Troubleshooting)?

### Description

Brackets reset language for a file type to text if the language set as default for the file type is defined by an extension and this extension doesn't define this file type in the list provided to language manager

### Steps to Reproduce

1. Download an extension which highlight a language not supported by Brackets (this [one](https://github.com/Athorcis/brackets-twig) by example)
2. Open a file which file type is not supported by the extension you installed
3. Set the language to the one supported by the extension
4. Set this language as the default one
5. Reload Brackets (with extensions).

**Expected behavior:** The language for the file is the one supported by the extension

**Actual behavior:** The language is Text.

### Versions
Windows 10 / Brackets 1.8
Also reproduced under Windows 7 / Brackets 1.8 (with only one extension the one mentioned above)

**Comment by haslam22** [comment]
Seems to be caused by `_updateFromPrefs` in `LanguageManager.js`. Languages defined by brackets extensions (like twig) aren't loaded straight away, so it will fail to find them when trying to load the preferences file for the first time. Further calls to `_updateFromPrefs` only look at changes to the preferences file so it won't pick up on the new languages. Adding a loop to check for invalid mappings fixes this issue:

```
        newNames.forEach(function(name) {
            var language  = _fileExtensionToLanguageMap[name];
            // Mapping returns undefined, but language exists
            if(!language && getLanguage(newMapping[name])) {
                // Add the extension to the language
                language = getLanguage(newMapping[name]);
                language[state.add](name);
            }
        });
```

**Comment by zaggino** [comment]
Fixed by https://github.com/adobe/brackets/pull/12979 to be released in 1.9
